art/estimators/object_detection/PyTorchFasterRCNN.py

```diff
@@ -112,20 +124,26 @@ def loss_gradient(self, x, y, **kwargs):
         :return: Loss gradients of the same shape as `x`.
         :rtype: `np.ndarray`
         """
+        import torch
         import torchvision
 
         self._model.train()
 
         # Apply preprocessing
         x, _ = self._apply_preprocessing(x, y=None, fit=False)
-        x = x.astype(np.uint8)
+
+        if y is not None:
+            for i, y_i in enumerate(y):
+                y[i]["boxes"] = torch.FloatTensor(y_i["boxes"]).to(self._device)
+                y[i]["labels"] = torch.LongTensor(y_i["labels"]).to(self._device)
+                y[i]["scores"] = torch.Tensor(y_i["scores"]).to(self._device)
 
         transform = torchvision.transforms.Compose([torchvision.transforms.ToTensor()])
 
         image_tensor_list = list()
 
         for i in range(x.shape[0]):
-            img = transform(x[i])
+            img = transform(x[i] / self.clip_values[1]).to(self._device)
```

Comment:
I observe that on several occasions you divide by `clip_values[1]`. This would break when a user provides clip values equal to `(min_clip, 0)` where `min_clip<0`. Would something like `np.abs(clip_values).max()` or `np.abs(clip_values[1] - clip_values[0]) / 2` be more senisble?

Reply:
You are right. The reason for this is that this estimator is model-specific for PyTorch's FasterRCNN which expects images in the pixel range [0, 1]. To enable inputs to `PyTorchFasterRCNN` in the range [0, 255] it normalises with `clip_values[1]`. I have now added assertions to prevent unexpected `clip_values` where where `min_clip!=0`.

Reply:
Awesome! Thanks for the clarification,  makes sense to me now +1: